sktime/forecasting/base/adapters/_lagllama.py

```diff
@@ -0,0 +1,626 @@
+- ensure interface compatibility by sktime.utils.estimator_checks.check_estimator
+- once complete: use as a local library, or contribute to sktime via PR
+- more details:
+  https://www.sktime.net/en/stable/developer_guide/add_estimators.html
+
+Mandatory implements:
+    fitting         - _fit(self, y, X=None, fh=None)
+    forecasting     - _predict(self, fh=None, X=None)
+
+Optional implements:
+    updating                    - _update(self, y, X=None, update_params=True):
+    predicting quantiles        - _predict_quantiles(self, fh, X=None, alpha=None)
+    OR predicting intervals     - _predict_interval(self, fh, X=None, coverage=None)
+    predicting variance         - _predict_var(self, fh, X=None, cov=False)
+    distribution forecast       - _predict_proba(self, fh, X=None)
+    fitted parameter inspection - _get_fitted_params()
+
+Testing - required for sktime test framework and check_estimator usage:
+    get default parameters for test instance(s) - get_test_params()
+"""
+# todo: write an informative docstring for the file or module, remove the above
+# todo: add an appropriate copyright notice for your estimator
+#       estimators contributed to sktime should have the copyright notice at the top
+#       estimators of your own do not need to have permissive or BSD-3 copyright
+
+__author__ = ["shlok191"]
+
+
+import subprocess
+
+from sktime.forecasting.base import BaseForecaster
+
+
+class _LagLlamaAdapter(BaseForecaster):
+    """Base class that interfaces the LagLlama forecaster.
+
+    Parameters
+    ----------
+    pretrained : bool, optional (default=True)
+        Dictates whether the pretrained LagLlama params should be initialized
+    """
+
+    _tags = {
+        "y_inner_mtype": "gluonts_PandasDataset_panel",
+        "X_inner_mtype": "gluonts_PandasDataset_panel",
+        "scitype:y": "both",
+        "ignores-exogeneous-X": True,
+        "requires-fh-in-fit": True,
+        "X-y-must-have-same-index": True,
+        "enforce_index_type": None,
+        "handles-missing-data": False,
+        "capability:insample": True,
+        "capability:pred_int": True,
+        "capability:pred_int:insample": True,
+        "authors": ["shlok191"],
+        "maintainers": ["shlok191"],
+        "python_version": None,
+        "python_dependencies": ["gluonts", "huggingface_hub", "lag_llama"],
+    }
+
+    def __init__(
+        self,
+        device=None,
+        context_length=None,
+        prediction_length=None,
+        num_samples=None,
+        batch_size=None,
+        nonnegative_pred_samples=None,
+        lr=None,
+        trainer_kwargs=None,
+    ):
+        import torch
+        from lag_llama.gluon.estimator import LagLlamaEstimator
+
+        # Defining private variable values
+        self.device = device
+        self.device_ = torch.device("cpu") if not device else device
+
+        self.context_length = context_length
+        self.context_length_ = 32 if not context_length else context_length
```

Comment:
please add the non mutable defaults as default in the signature of the init method.

Reply:
Got it!